README.md

```diff
@@ -169,6 +192,10 @@ Earlier versions of this project were forked from the [miketheman heroku-buildpa
 
 Make sure you have `DD_DYNO_HOST` set to `true` and that `HEROKU_APP_NAME` has a value set for every Heroku application. See the [Hostname section](#hostname) for details.
 
+### After upgrading the buildpack or the agent, the agent is reporting errors when starting up
+
+After an upgrade of the buildpack or the agent, a full recompilation of your application's slug, with a clean build cache, is required. Check [the upgrading and slug recompilation section](#upgrading-and-slug-recompilation) for details.
```

Comment:
Remember for documentation we want to be direct: "After an upgrade of the buildpack or agent, you must clear your build cache and recompile your application's slug."

Also, let's be sure that we always reference clearing cache before build/recompile, since that's the correct order. It should be obvious to most people, but for someone not familiar with things, they might try to recompile, then clear cache.